libvast/test/module.cpp

```diff
@@ -0,0 +1,689 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2022 The VAST Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#define SUITE module
+
+#include "vast/module.hpp"
+
+#include "vast/aliases.hpp"
+#include "vast/concept/convertible/data.hpp"
+#include "vast/data.hpp"
+#include "vast/legacy_type.hpp"
+#include "vast/test/test.hpp"
+#include "vast/type.hpp"
+
+#include <caf/error.hpp>
+#include <caf/sum_type.hpp>
+#include <caf/test/dsl.hpp>
+#include <fmt/format.h>
+
+#include <optional>
+#include <string_view>
+
+using namespace vast;
+
+caf::expected<type>
+to_type(const std::vector<type>& known_types, const data& declaration,
+        std::string_view name = "");
+
+caf::expected<record_type>
+to_record(const std::vector<type>& known_types,
+          const std::vector<vast::data>& record_list) {
```

Comment:
```suggestion
caf::expected<record_type>
to_record(const std::vector<type>& known_types,
          const list& field_declarations) {
```

`list` is `std::vector<data>`, so please use that for clarity.

I think the name `field_declarations` makes sense for the argument if we're using "declaration" as the term for the relevant subsection of the YAML.

Reply:
Please also do a general pass over the function and try to give appropriate variable names.

Reply:
I have created a [railroad diagram](https://media.app.shortcut.com/api/attachments/files/clubhouse-assets/59a979b5-168d-459a-ad4b-fe6d2b009b9e/6286118c-351e-4400-b00e-a909a44677e1/diagram.xhtml) that helps in reviewing variable names. Reviewing variable names makes more sense in the "yaml-modules" PR, though, where these names will become final.

Reply:
That makes sense.